### PR TITLE
Dev platform aks

### PR DESCRIPTION
**Pull request by saliceti**
## Context
Make it easier to deploy Apply review apps to dev clusters to be able to test clusters quickly

## Changes proposed in this pull request
Create environment configuration `dev_platform_aks` to be able to deploy Apply to dev clusters

Other improvements:
- Update paas terraform to 1.3.5
- Pin kubernetes module
- Display URL in terraform output

## Guidance to review
- create a dev cluster clusterX
- run `make dev_platform_review_aks deploy PR_NUMBER=2222 CLUSTER=clusterX PASSCODE=1`
- deployment to paas still working: https://apply-review-7917.london.cloudapps.digital/candidate/account

## Link to Trello card
https://trello.com/c/ymDMSY1e

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by saliceti** [comment]
[Deploy Apply to dev clusters](https://trello.com/c/ymDMSY1e/141-deploy-apply-to-dev-clusters)

File: Makefile [hunk 1]
```diff
@@ -196,7 +205,7 @@ deploy-init:
 	$(eval export TF_VAR_paas_docker_image=ghcr.io/dfe-digital/apply-teacher-training:$(IMAGE_TAG))
 
 	az account set -s $(AZURE_SUBSCRIPTION) && az account show
-	terraform -chdir=terraform/$(PLATFORM) init -reconfigure -backend-config=./workspace_variables/$(APP_ENV)_backend.tfvars $(backend_key)
+	terraform -chdir=terraform/$(PLATFORM) init -reconfigure -upgrade -backend-config=./workspace_variables/$(APP_ENV)_backend.tfvars $(backend_key)
```
Comment (smneal) [inline Makefile:208]: Should we remove the `-upgrade`?

Comment (saliceti) [reply Makefile:208]: We've pinned the provider versions exactly so there is no risk of upgrading to an unknown version

**Review by smneal — APPROVED** [review]
Nice work

**Pull request merged**